reactioncommerce/reaction
# 1780 product archive

Resolves #1780 
Resolves #1799 

Prevent instances where archiving a product / variant would also archive products
that are already archived, resulting in more revision documents being
created when not necessary.

**To Test**

0. From new product with top level and child variants
1. Archive product (very top level using the toolbar button)
2. Attempt to restore top level variant
3. Should receive an error that you cannot restore
4. Attempt to restore child variant
5. Should receive error that you cannot restore
6. Attempt to create a top level variant
7. Should receive error that you cannot create
8. Attempt to create a child variant
9. Should receive error that you cannot create


**tl;dr**

When a parent product or variant is deleted, you should not be able to add or restore the children because the dependency tree will be broken on publish.
